src/Utility/FieldFunctions.C

```diff
@@ -317,6 +317,54 @@ Real FieldFunction::evaluate (const Vec3& X) const
 }
 
 
+Vec3 FieldFunction::gradient (const Vec3& X) const
+{
+  if (pidx >= field.size() || !field[pidx])
+    return Vec3();
+
+  const Vec4* x4 = dynamic_cast<const Vec4*>(&X);
+  if (!x4)
+    return Vec3();
+
+  int level = this->findClosestLevel(x4->t);
+  if (level < 0) return Vec3();
+  if (level != currentLevel)
+    if (const_cast<FieldFunction*>(this)->load({fName},bName,level))
+      currentLevel = level;
+
+  Vector res;
+  field[pidx]->gradFE(ItgPoint(x4->u, x4->idx), res);
+  return res;
+}
+
+
+SymmTensor FieldFunction::hessian (const Vec3& X) const
+{
+  if (pidx >= field.size() || !field[pidx])
+    return SymmTensor(3);
+
+  const Vec4* x4 = dynamic_cast<const Vec4*>(&X);
+  if (!x4)
+    return SymmTensor(3);
+
+  int level = this->findClosestLevel(x4->t);
+  if (level < 0) return SymmTensor(3);
+  if (level != currentLevel)
+    if (const_cast<FieldFunction*>(this)->load({fName},bName,level))
+      currentLevel = level;
+
+  Matrix hess;
+  field[pidx]->hessianFE(ItgPoint(x4->u), hess);
+
+  SymmTensor result(hess.rows());
+  for (size_t d1 = 1; d1 <= hess.rows(); ++d1)
+    for (size_t d2 = d1; d2 <= hess.cols(); ++d2)
+      result(d1,d2) = hess(d1,d2);
+
+  return result;
```

Comment:
Same, consider std::move() ?

Reply:
Actually move here is hindering copy ellision so I have no idea why I added them in the first place. I removed the others instead.

Reply:
Right, think I remember seeing similar issues when we run some Coverty scanning on the FEDEM code earlier. But as long we are consistent I'm happy with it

Reply:
FYI, you added std::move() one more place in the same commit 9dd8bbc23c5befc25b658995f981c483d59fa539

Reply:
hmm, this must have been some (faulty) static analyzer output. i'll remove this one as well (won't bother to rerun jenkins).